src/app/ReadClient.h

```diff
@@ -284,6 +284,11 @@ class ReadClient : public Messaging::ExchangeDelegate
          * things like min/max intervals based on the session parameters).
          */
         virtual void OnCASESessionEstablished(const SessionHandle & aSession, ReadPrepareParams & aSubscriptionParams) {}
+
+        /*
+         * return the internal fatal error in callback
+         */
+        virtual CHIP_ERROR CheckInternalError() { return CHIP_NO_ERROR; }
```

Comment:
The name is `Check` and description says `returns the internal....`.

Could we rename as this does not check anything? Something like `GetProcessingError` or `GetLastError` and update the description explaining when this is set and what it represents. the `internal fatal error` is not as clear.

Also this seems to be `const`